minikube/src/CommandCluster/CommandCluster.tsx

```diff
@@ -0,0 +1,177 @@
+    }, 500);
+
+    return () => clearInterval(intervalId);
+  }, []);
+
+  React.useEffect(() => {
+    if (startOpen) {
+      setOpenDialog(true);
+    }
+  }, [startOpen]);
+
+  function handleRunCommand({ clusterName, driver }: { clusterName: string; driver: string }) {
+    if (DEBUG) {
+      console.log('runFunc handleSave', clusterName);
+    }
+    setTheCluster(clusterName);
+    setActing(true);
+    onCommandStarted && onCommandStarted();
+
+    function runFunc(clusterName: string) {
+      if (DEBUG) {
+        console.log('runFunc', clusterName);
+      }
+      const args = [command, '-p', clusterName];
+      if (driver) {
+        args.push('--driver', driver);
+      }
+      const minikube = runCommand('minikube', args, {});
+      if (DEBUG) {
+        console.log('runFunc after runCommand');
+      }
+      const stdoutData: string[] = [];
+      const errorData: string[] = [];
+      setRunning(true);
+
+      minikube.stdout.on('data', data => {
+        if (DEBUG) {
+          console.log('runFunc stdout:', data);
+        }
+        stdoutData.push(data);
+        allDataRef.current.push(data);
+
+        if (data.includes(finishedText)) {
+          setCommandDone(true);
+        }
+      });
+      minikube.stderr.on('data', (data: string) => {
+        if (DEBUG) {
+          console.log('runFunc stderr:', data);
+        }
+        errorData.push(data);
+        allDataRef.current.push(data);
+      });
+      minikube.on('exit', code => {
+        if (DEBUG) {
+          console.log('runFunc exit code:', code);
+        }
+        setCommandDone(true);
+        // if (code === 0) {
+        //   resolve({ stdoutData, errorData });
+        // } else {
+        //   reject({ stdoutData, errorData });
+        // }
+      });
+
+      onConfirm();
+      if (DEBUG) {
+        console.log('runFunc finished');
+      }
+    }
+    if (DEBUG) {
+      console.log('runFunc dispatching', clusterName);
+    }
+
+    clusterAction(() => runFunc(clusterName), {
+      startMessage: `About to "${command}" cluster "${clusterName}"…`,
+      cancelledMessage: `Cancelled "${command}" cluster "${clusterName}".`,
+      successMessage: `Cluster "${command}" of "${clusterName}" begun.`,
+      errorMessage: `Failed to "${command}" ${clusterName}.`,
+
+      cancelCallback: () => {
+        setActing(false);
+        setRunning(false);
+        handleClose();
+        setOpenDialog(false);
+      },
+    });
+  }
+
+  return (
+    <CommandDialog
+      open={openDialog}
+      onClose={() => {
+        setOpenDialog(false);
+        handleClose();
+        allDataRef.current = [];
+        setActing(false);
+        setCommandDone(false);
+      }}
+      onConfirm={({ clusterName, driver }) => handleRunCommand({ clusterName, driver })}
+      command={command}
+      title={
+        askClusterName
+          ? acting
+            ? `Creating Minikube Cluster ${theCluster}`
+            : 'New Minikube Cluster'
```

Comment:
maybe this would be better as a sort-of description? what do you think
```suggestion
            : 'Create a new Minikube cluster.'
```

Reply:
done